bookserver/crud.py

```diff
@@ -118,10 +127,118 @@ async def fetch_course(course_name: str) -> CoursesValidator:
     return CoursesValidator.from_orm(res.scalars().first())
 
 
+async def create_course(course_info: CoursesValidator) -> CoursesValidator:
+    new_course = Courses(**course_info.dict())
+    async with async_session() as session:
+        res = session.add(new_course)
+        await session.commit()
+
+    return CoursesValidator.from_orm(res) if res else None
+
+
+# auth_user
+# ---------
 async def fetch_user(user_name: str) -> AuthUserValidator:
     query = select(AuthUser).where(AuthUser.username == user_name)
     async with async_session() as session:
         res = await session.execute(query)
         rslogger.debug(f"res = {res}")
     user = res.scalars().first()
     return AuthUserValidator.from_orm(user) if user else None
+
+
+async def create_user(user: AuthUserValidator) -> int:
+    """
+    The given user will have the password in plain text.  First we will hash
+    the password then add this user to the database.
+    """
+    new_user = AuthUser(**user.dict())
+    print(settings.web2py_private_key)
+    crypt = CRYPT(key=settings.web2py_private_key, salt=True)
+    new_user.password = str(crypt(user.password)[0])
```

Comment:
Does this work? The format is `alg$salt$hash`, and here I only see the hash being generated.

Reply:
yes it works.
the salt is part of the password that is stored in the database. 

the call to crypt returns the full thing: `alg$salt$hash$

Reply:
I've been developing and testing logging in as testuser1 with no problems 😄

Reply:
Great!